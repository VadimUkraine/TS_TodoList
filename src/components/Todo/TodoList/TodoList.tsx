import React, { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import List from "@material-ui/core/List";
import { RootState } from "../../../redux/reducers/rootReducer";
import { deleteTodoRequest, changeTodoRequest } from "../../../redux/actions/todo";
import TodoListItem from "./TodoListItem";
import { Todo } from '../../../types';

export const TodoList: React.FC = () => {

  const dispatch = useDispatch();
  const list = useSelector((state: RootState) => state.todo.list);

  const [editID, setEditId] = useState("");

  const handleDelete = (id: string) => {
    dispatch(deleteTodoRequest(id));
  };

  const handleSetEditId = (id: string) => {
    setEditId(id);
  };

  const handleChangeTodo = (id: string, text: string) => {
    dispatch(changeTodoRequest(id, text));
  };

  return (
    <List data-testid={"todo-list-component"}>
      {list.map((todo: Todo) => (
        <TodoListItem
          key={todo.id}
          todo={todo}
          deleteTodo={handleDelete}
          editID={editID}
          setEditId={handleSetEditId}
          changeTodo={handleChangeTodo}
        />
      ))}
    </List>
  );
};

export default TodoList;
