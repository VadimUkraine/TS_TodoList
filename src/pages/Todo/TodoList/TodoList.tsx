
import React from "react";
import List from "@material-ui/core/List";
import { useDispatch, useSelector } from "react-redux";
import { RootState } from "../../../rootReducer";
import { deleteTodoRequest } from "../actions";
import TodoListItem from "../../../components/TodoListItem";

export const TodoList: React.FC = () => {

  const dispatch = useDispatch();
  const list = useSelector((state: RootState) => state.todo.list);

  const handleDelete = (todo: string) =>{
    dispatch(deleteTodoRequest(todo))
  }



  return(
    <List >
      {list.map((todo: string, index: number)=>(
        <TodoListItem key={index} todo={todo} deleteTodo={handleDelete}   />
      ))}
    </List>
  )
}