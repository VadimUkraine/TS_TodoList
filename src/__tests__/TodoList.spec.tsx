import React from 'react';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import {
  render, screen, cleanup,
} from '@testing-library/react';
import TodoList from '../components/Todo/TodoList';
import configureStore from '../redux/store';

const mockStore = {
  todo: {
    list: [
      {
        id: '1c0573e9-2e59-48c8-9b0b-ec5f63afbe0d',
        text: 'buy beer',
        date: '04.11.2020, 17:47',
      },
      {
        id: 'a1f80f52-1690-4c20-a74b-32ac420ece61',
        text: 'buy bread',
        date: '04.11.2020, 13:30',
      },
    ],
  },
};

const store = configureStore(mockStore);

beforeEach(() => render(
  <Provider store={store}>
    <TodoList />
  </Provider>,
));

afterEach(cleanup);

test('it renders component', () => {
  expect(screen.getAllByRole('listitem')).toHaveLength(2);
});

test('it renders correctly todoList', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <TodoList />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
