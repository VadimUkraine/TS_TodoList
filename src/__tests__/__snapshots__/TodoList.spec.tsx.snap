// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly todoList 1`] = `
<ul
  className="MuiList-root MuiList-padding"
  data-testid="todo-list-component"
>
  <li
    className="MuiListItem-container"
  >
    <div
      className="MuiListItem-root makeStyles-todoItem-11 MuiListItem-gutters MuiListItem-secondaryAction"
      data-testid="todo-list-item"
      disabled={false}
    >
      <div
        className="makeStyles-todoItemContent-8"
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary makeStyles-textStyles-12 MuiTypography-body1 MuiTypography-displayBlock"
          >
            buy beer
          </span>
        </div>
        <span
          className="MuiTypography-root makeStyles-dateStyles-13 MuiTypography-caption MuiTypography-gutterBottom MuiTypography-displayBlock"
        >
          04.11.2020, 17:47
        </span>
      </div>
    </div>
    <div
      className="MuiListItemSecondaryAction-root makeStyles-todoButtonsWrap-9"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Edit
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btnDelete-10 MuiButton-containedSecondary MuiButton-containedSizeSmall MuiButton-sizeSmall"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Delete
        </span>
      </button>
    </div>
  </li>
  <li
    className="MuiListItem-container"
  >
    <div
      className="MuiListItem-root makeStyles-todoItem-11 MuiListItem-gutters MuiListItem-secondaryAction"
      data-testid="todo-list-item"
      disabled={false}
    >
      <div
        className="makeStyles-todoItemContent-8"
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary makeStyles-textStyles-12 MuiTypography-body1 MuiTypography-displayBlock"
          >
            buy bread
          </span>
        </div>
        <span
          className="MuiTypography-root makeStyles-dateStyles-13 MuiTypography-caption MuiTypography-gutterBottom MuiTypography-displayBlock"
        >
          04.11.2020, 13:30
        </span>
      </div>
    </div>
    <div
      className="MuiListItemSecondaryAction-root makeStyles-todoButtonsWrap-9"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Edit
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btnDelete-10 MuiButton-containedSecondary MuiButton-containedSizeSmall MuiButton-sizeSmall"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Delete
        </span>
      </button>
    </div>
  </li>
</ul>
`;
